src/reportengine/configparser.py

```diff
@@ -132,6 +133,78 @@ def f_(self, val, *args, **kwargs):
 
     return f_
 
+def parse_record_(self, val):
+    """Template function for parsing <key>_recorded_spec_ from lockfile"""
+    return val
+
+def record_from_defaults(f):
+    f"""Decorator for recording default values. Given a key, for example
```

Comment:
This doesn't work, unfortunately

```python
In [1]: a = 5                                                                                                         

In [2]: def f(): 
   ...:     f"""a is {a} but this isn't __doc__""" 
   ...:                                                                                                               

In [3]: f.__doc__
```

Reply:
ah crap, I will fix this once we have stable tokens